addons/event_sale/wizard/event_batch_configurator.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class BatchEventConfigurator(models.TransientModel):
+    _name = 'event.event.configurator.batch'
+    _description = 'Event batch Configurator'
+
+    configurable_event_ids = fields.One2many('event.editor.line', 'editor_id')
+    sale_order_id = fields.Many2one('sale.order', 'Sales Order', required=True, ondelete='cascade')
+
+    @api.model
+    def default_get(self, fields):
+        res = super(BatchEventConfigurator, self).default_get(fields)
+        if not res.get('sale_order_id'):
+            sale_order_id = res.get('sale_order_id', self._context.get('active_id'))
+            res['sale_order_id'] = sale_order_id
+        sale_order = self.env['sale.order'].browse(res.get('sale_order_id'))
+        configurable_event_ids = []
+        for so_line in [l for l in sale_order.order_line if l.product_type == 'event']:
+            for _ in range(int(so_line.product_uom_qty)):
+                configurable_event_ids.append([0, 0, {
+                    'event_id': so_line.event_id.id,
+                    'event_ticket_id': so_line.event_ticket_id.id,
+                    'product_id': so_line.product_id,
+                    'order_id': sale_order,
+                    'sale_order_line_id':so_line,
+                }])
+        res['configurable_event_ids'] = configurable_event_ids
+        res = self._convert_to_write(res)
+        return res
+
+    def action_configure(self):
+        self.ensure_one()
+        for configurable_event_id in self.configurable_event_ids:
+            configurable_event_id.get_event_data()
+        return {'type': 'ir.actions.act_window_close'}
+
+class EventConfiguratorLine(models.TransientModel):
```

Comment:
can we move this model in a different file ?

Reply:
We can, but then we would want to move the model from event_edit_registration.py also, otherwise I think it won't really be consistent :/

Reply:
yes we can move it as well :)